flytekit/image_spec/image_spec.py

```diff
@@ -67,6 +67,7 @@ class ImageSpec:
 
     def __post_init__(self):
         self.name = self.name.lower()
+        self._is_force_push = False  # False by default
```

Comment:
Do we need to update:

https://github.com/flyteorg/flytekit/blob/632b2c2d563146801d218dece4a1b58b9fb26d38/flytekit/image_spec/image_spec.py#L248

such that the `force_push` field is not a part of the hash?